lint/rule_template_datasource.go

```diff
@@ -2,47 +2,95 @@ package lint
 
 import (
 	"fmt"
+	"sort"
+	"strings"
 )
 
+// This rule checks that the dashboard has single templated datasource
+func (d *Dashboard) checkPrometheusOrLokiDS(template *Template) Result {
+
+	if template.Name != "datasource" {
+		return Result{
+			Severity: Error,
+			Message:  fmt.Sprintf("Dashboard '%s' templated datasource variable named '%s', should be named 'datasource'", d.Title, template.Name),
+		}
+	}
+
+	if template.Label != "Data Source" {
+		return Result{
+			Severity: Error,
+			Message:  fmt.Sprintf("Dashboard '%s' templated datasource variable labeled '%s', should be labeled 'Data Source'", d.Title, template.Label),
+		}
+	}
+
+	if template.Query != Prometheus && template.Query != Loki {
+		return Result{
+			Severity: Error,
+			Message:  fmt.Sprintf("Dashboard '%s' templated datasource variable query is '%s', should be 'prometheus' or 'loki'", d.Title, template.Query),
+		}
+	}
+	return ResultSuccess
+}
+
+func (d *Dashboard) checkPrometheusAndLokiDS(datasources []Template) Result {
+
+	// move loki to 0 index
+	sort.SliceStable(datasources, func(i, j int) bool {
+		return datasources[i].Query < datasources[j].Query
+	})
+
+	if !(datasources[0].Query == Loki && datasources[1].Query == Prometheus) {
+		return Result{
+			Severity: Error,
+			Message:  fmt.Sprintf("Dashboard '%s' with 2 templated datasources should have 'prometheus' and 'loki' types", d.Title),
+		}
+	}
```

Comment:
Similar to the previous comment. It's "okay" if a dashboard has two (or more) datasources which are not loki and prometheus. The dashboard linter would simply have no opinion/rule in this case.